src/util/fs.js

```diff
@@ -144,21 +144,29 @@ async function buildActionsForCopy(
       srcFiles = await readdir(src);
     }
 
-    if (await exists(dest)) {
-      const destStat = await lstat(dest);
+    let destStat;
+    try {
+      // try accessing the destination
+      destStat = await lstat(dest);
+    } catch (e) {
+      // proceed if destination doesn't exist, otherwise error
+      if (e.code !== 'ENOENT') throw e;
+    }
 
+    // if destination exists
+    if (destStat) {
       const bothSymlinks = srcStat.isSymbolicLink() && destStat.isSymbolicLink();
       const bothFolders = srcStat.isDirectory() && destStat.isDirectory();
       const bothFiles = srcStat.isFile() && destStat.isFile();
 
-      if (srcStat.mode !== destStat.mode) {
+      // EINVAL access errors sometimes happen which shouldn't because node shouldn't be giving
+      // us modes that aren't valid. investigate this, it's generally safe to proceed.
+
+      /* if (srcStat.mode !== destStat.mode) {
```

Comment:
This commit was added by @kittens in https://github.com/yarnpkg/yarn/commit/4bbc653673bd65caf29084a30428cfa634166a66.
I'll ping him to give some context